### PR TITLE
MAGN-9813 API Documentation Priority 2 (Common PR)

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

Documentation got added to Priority 2 classes/interfaces.
The PR contains changes from the PR of @pbidenko - https://github.com/DynamoDS/Dynamo/pull/6407

### Declarations

- [ ] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps

### CODE REVIEW

**Comment by marimano** [comment]
Hi @ramramps, this PR needs completed documentation for 18 more classes which I will add when @pbidenko finishes.
But the PR is enormous and you can start reviewing it little by little if you would like to :)

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt [hunk 1]
```diff
@@ -71,4 +71,4 @@ int MajorVersion = 1;
 int MinorVersion = 0;
 int BuildNumber = 1;
 int RevisionNumber = ((int)(DateTime.UtcNow - new DateTime(2016,1,1)).TotalDays)*10+((int)DateTime.UtcNow.Hour)/3;
-#>
+#>
```
Comment (marimano) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt]: redundant new line symbol causes build fail

**Comment by marimano** [comment]
the PR is ready for review

File: src/DynamoCore/Configuration/Configurations.cs [hunk 2]
```diff
@@ -3,58 +3,140 @@
...
+    /// <summary>
+    ///     This class contains properties that are used in Dynamo.
+    /// </summary>
     public class Configurations
     {
         #region Dynamo Universal Constants
 
-        // Add 0.5 to place the point in the middle of a pixel to sharpen it
+        /// <summary>
+        ///     User interface culture
```
Comment (ramramps) [inline src/DynamoCore/Configuration/Configurations.cs]: White space. See below.

File: src/DynamoCore/Configuration/Configurations.cs [hunk 3]
```diff
@@ -3,58 +3,140 @@
...
+        /// </summary>
         public static readonly double DoubleSliderTextBoxWidth = 55.0;
+
+        /// <summary>
+        ///     Default width of Integer Slider
+        /// </summary>
         public static readonly double IntegerSliderTextBoxWidth = 30.0;
+
+        /// <summary>
+        ///     Maximal width of Watch Node
```
Comment (ramramps) [inline src/DynamoCore/Configuration/Configurations.cs]: Maximal = Maximum

File: src/DynamoCore/Configuration/Configurations.cs [hunk 4]
```diff
@@ -3,58 +3,140 @@
 
 namespace Dynamo.Configuration
 {
+    /// <summary>
+    ///     This class contains properties that are used in Dynamo.
+    /// </summary>
     public class Configurations
     {
```
Comment (ramramps) [inline src/DynamoCore/Configuration/Configurations.cs]: Question : Why the configurations are public? I guess these should be internal or protected. Why the API users need to know about his?

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 5]
```diff
@@ -19,19 +19,47 @@ namespace Dynamo.Configuration
...
-        public const int DefaultMaxNumRecentFiles = 10;
-        public static string DynamoTestPath = null;
         private string numberFormat;
         private string lastUpdateDownloadPath;
         private int maxNumRecentFiles;
+
+        /// <summary>
+        /// Indicates the maximum number of files shown in Recent Files
+        /// </summary>
+        public const int DefaultMaxNumRecentFiles = 10;
```
Comment (ramramps) [inline src/DynamoCore/Configuration/PreferenceSettings.cs]: Why this is not internal?

File: src/DynamoCore/Core/CrashPromptArgs.cs [hunk 6]
```diff
@@ -12,12 +15,33 @@ public enum DisplayOptions
             HasFilePath = 0x00000004
         }
 
+        /// <summary>
+        ///     Flag which indicates whether args contain default text overriden,
```
Comment (ramramps) [inline src/DynamoCore/Core/CrashPromptArgs.cs]: Returns cref = Dispalyoptions. Ex:....

File: src/DynamoCore/Core/CrashPromptArgs.cs [hunk 7]
```diff
@@ -12,12 +15,33 @@ public enum DisplayOptions
...
+        ///     has details or has file path
+        /// </summary>
         public DisplayOptions Options { get; private set; }
+
+        /// <summary>
+        ///     Details of crash
```
Comment (ramramps) [inline src/DynamoCore/Core/CrashPromptArgs.cs]: Returns Details of a crash.

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 8]
```diff
@@ -12,6 +12,17 @@
 
 namespace Dynamo.Extensions
 {
+    /// <summary>
+    /// Provides functionality for loading Dynamo's extensions.
+    /// This class loads particulary formatted XMLs which contain information about
```
Comment (ramramps) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: Remove : particulary

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 9]
```diff
@@ -32,6 +43,11 @@ private IExtension Load(ExtensionDefinition extension)
             }
         }
 
+        /// <summary>
+        /// Loads <see cref="IExtension"/> from assembly by passed path
```
Comment (ramramps) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: Remove : by passed path

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 10]
```diff
@@ -84,6 +105,9 @@ public IEnumerable<IExtension> LoadDirectory(string extensionsPath)
             return result;
         }
 
+        /// <summary>
+        /// Logs messages
```
Comment (ramramps) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: This event is used for logging messages.

**Comment by ramramps** [comment]
@marimano - Add RETURNS to all Properties. Please check the Documentation for Priority 1 and 3.

**Comment by ramramps** [comment]
Merge this.

**Comment by ramramps** [comment]
@marimano = Few changes.

**Pull request merged**